components/devtools/actors/watcher.rs

```diff
@@ -429,4 +431,31 @@ impl WatcherActor {
             },
         }
     }
+
+    pub fn emit_will_navigate(
+        &self,
+        browsing_context_id: BrowsingContextId,
+        url: ServoUrl,
+        connections: &mut Vec<TcpStream>,
+    ) {
+        let msg = serde_json::json!({
+            "browsingContextID": browsing_context_id,
```

Comment:
We will need to use browsing_context_id.0 to get the id number instead of the string representation.